src/components/HdToggle.vue

```diff
@@ -0,0 +1,243 @@
+<template>
+  <div
+    :class="{
+      'hd-toggle--is-open': open,
+      'hd-toggle--is-using-mouse': isUsingMouse,
+    }"
+    class="hd-toggle"
+    @keydown="setUsingMouse(false)"
+    @mousedown="setUsingMouse(true)"
+  >
+    <button
+      :disabled="!canBeToggled"
+      class="hd-toggle__control"
+      type="button"
+      @click="toggleOpen"
+    >
+      {{ title }}
+    </button>
+    <div
+      ref="body"
+      :style="{
+        maxHeight: `${maxHeight}px`,
+        transitionDuration: `${transitionDuration}ms`,
+      }"
+      class="hd-toggle__body"
+    >
+      <slot />
+    </div>
+  </div>
+</template>
+
+<script>
+import { OnResizeService } from 'homeday-blocks';
+
+export default {
+  name: 'HdToggle',
+  props: {
+    title: {
+      type: String,
+      default: '',
+    },
+    canBeToggled: {
```

Comment:
Can you please add a test covering this use case as well? I think should be tested, as it's external API of the component.